Examples/ImportCategoriesAndProducts.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace IReportsApiExamples.Examples
+{
+    public class ImportCategoriesAndProducts
+    {
+        /// <summary>Reads the categories and products from specified Json files and uploads them to the server</summary>
+        /// <param name="wrapper">The ApiWrapper object to use</param>
+        /// <param name="categoriesFilePath">The path to the categories Json file</param>
+        /// <param name="categoriesFilePath">The path to the products Json file</param>
+        /// <param name="libraryCode">The code of the desired library to put the categories and products in</param>
+        public static async Task DoWork(
+            ApiWrapper wrapper,
+            string categoriesFilePath,
+            string productsFilePath,
+            string libraryCode)
+        {
+            var categoryList = GetListFromJson<CategoryModel>(categoriesFilePath);
+            var productList = GetListFromJson<ProductMetadataModel>(productsFilePath);
+
+            await UploadCategories(wrapper, categoryList);
+            await UploadProducts(wrapper, productList, libraryCode);
+        }
+
+        /// <summary>Uploads the products first. Then uploads the extended metadata, authors and the list
+        /// of categories the product is in, if it the product model includes them</summary>
+
+        /// <param name="wrapper">The ApiWrapper object to use</param>
+        /// <param name="productList">The list of deserialized products from the Json file</param>
+        /// <param name="libraryCode">The code of the desired library to put the products in</param>
+        private static async Task UploadProducts(ApiWrapper wrapper,
```

Comment:
Put this parameter on a new line.
You could also consider a name like PostProducts. Just because Upload kinda sounds like it's a video file or something.